fastfield_codecs/src/lib.rs

```diff
@@ -19,10 +21,32 @@ pub trait FastFieldCodecReader: Sized {
     fn max_value(&self) -> u64;
 }
 
+pub trait FastFieldCodecReaderU128: Sized {
+    /// reads the metadata and returns the CodecReader
+    fn open_from_bytes(bytes: &[u8]) -> std::io::Result<Self>;
+
+    /// Get value for doc
+    fn get(&self, doc: u64, data: &[u8]) -> Option<u128>;
+
+    /// Iterator
+    ///
+    /// Replace with opaque type after: https://github.com/rust-lang/rust/issues/63063
+    fn iter<'a>(&'a self, data: &'a [u8]) -> Box<dyn Iterator<Item = Option<u128>> + 'a>;
+
+    /// Get positions (=docs in single value) for provided value range
+    fn get_between_vals(&self, range: RangeInclusive<u128>, data: &[u8]) -> Vec<usize>;
```

Comment:
Why not return a `Vec<u32>` ?

Reply:
It can be used in a multivalued field and there we support more than u32. In that case are the returned values not docids

Reply:
I see... I suspect we'll have to make it u32 in the end. Well, we will see.